jib-core/src/test/java/com/google/cloud/tools/jib/frontend/JavaLayerConfigurationsTest.java

```diff
@@ -379,4 +392,65 @@ public void testAddDirectoryContents_complex() throws IOException {
             "/base/test/D.class"),
         configurations.getExtraFilesLayerEntries());
   }
+
+  @Test
+  public void testAddExtraDirectoryContents() throws IOException {
+    temporaryFolder.newFolder("src", "main", "jib");
+    temporaryFolder.newFile("src/main/jib/fileA");
+    temporaryFolder.newFile("src/main/jib/fileB");
+    temporaryFolder.newFile("src/main/jib/fileC");
+    temporaryFolder.newFolder("src", "main", "jib", "folder");
+    temporaryFolder.newFile("src/main/jib/folder/fileD");
+    temporaryFolder.newFile("src/main/jib/folder/fileE");
+    temporaryFolder.newFile("src/main/jib/folder/folder2");
```

Comment:
Perhaps `newFile` --> `newFolder`. In that case, calling `newFolder("src", "main", "jib", "folder", "folder2")` will be sufficient (no need for multiple `newFolder()` calls).